src/core/interaction/pickingcontroller.cpp

```diff
@@ -91,100 +91,106 @@ void PickingController::propagateEvent(Event* event, EventPropagator* propagator
 void PickingController::propagateEvent(TouchEvent* e, EventPropagator* propagator) {
     auto& touchPoints = e->touchPoints();
 
-    std::unordered_map<size_t, std::vector<TouchPoint>> pickngIdToTouchPoints;
-    std::unordered_map<size_t, const PickingAction*> pickingIdToAction;
     std::vector<int> usedPointIds;
 
-    for (auto& point : touchPoints) {
-        const auto coord = glm::clamp(point.pos(), dvec2(0.0), dvec2(e->canvasSize() - uvec2(1)));
-        if (point.state() == TouchState::Started) {
-            auto pa = findPickingAction(coord);
-            tstate_.pointIdToPickingId[point.id()] = pa;
-            tstate_.pickingIdToPressNDC[pa.index] = point.ndc();
+    const auto pickId = [&](size2_t coord) -> size_t {
+        if (auto src = src_.lock()) {
+            const auto value = src->readPixel(coord, LayerType::Picking);
+            if (value.a > 0.0) {
+                return PickingManager::colorToIndex(uvec3(value));
+            }
         }
+        return 0;
+    };
 
-        auto it = tstate_.pointIdToPickingId.find(point.id());
-        auto pa = it != tstate_.pointIdToPickingId.end() ? it->second : findPickingAction(coord);
+    // Strategy for treating touch points for picking objects
+    //
+    // One touch event per initially picked object.
+    //  + Single picking event per object
+    //  + Picked objects will have access to multiple touch points
+    //  - Points initially picking an object may split to multiple object: tricky to maintain state
+    //  (ignored here)
+    //
+    // Other alternative considered:
+    // Treat each touch point individually.
+    //  + Easy to track state
+    //  - Multiple picking events (Enter/Leave/Press) per object possible
+    //  - Picked objects will not have access to multiple touch points
+    //
+
+    // Strategy: One touch event per initially picked object.
+    std::unordered_map<size_t, std::vector<TouchPoint>> pickingIdToTouchPoints;
```

Comment:
Consider `using` in the class definition with a name for the map type.

Reply:
I'm not too fond of hiding types like that with using.. especially not at the declaration site of the member. 
I really want to be able to easily see all the types of all the members in a class. I says _alot_ about a class. And if I have to go through an indirection with some using, that's often not even at the same place in the code, that is a huge pain...